projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.html

```diff
@@ -2,15 +2,19 @@
 
     <!-- dsp-resource-representation -->
     <div class="center">
-        <dsp-still-image *ngIf="stillImageRepresentations" [images]="stillImageRepresentations">
+        <dsp-still-image *ngIf="stillImageRepresentations.length" [images]="stillImageRepresentations">
```

Comment:
could this ever lead to a "Cannot read property 'length' of undefined" error?

Reply:
I reset the stillImageRepresentation always as empty array before getting the data. So it shouldn't end in the mentioned error. But we have to improve this representation-thing anyway. As soon we'll have other representation it could be messy. At the moment I do not have better solution.

Reply:
Is this related to change detection? In order to trigger change detection for a complex input, a new object / array has to be assigned.

Reply:
have you tried this:

https://github.com/dasch-swiss/dsp-ui-lib/blob/ce738efb2dcf65f9ecd2cbf583241cd49968cd94/projects/dsp-ui/src/lib/viewer/representation/still-image/still-image.component.ts#L157-L167

Reply:
It's in case you switch between a resource with fileRepresentation and a resource without. In one case we have to display `<dsp-still-image>` and in the other we don't have to.

Reply:
> I reset the stillImageRepresentation always as empty array before getting the data.

So I guess this is correct. I assume this is done in `ngOnChanges` when the resource Iri Input changes, right?

Reply:
exactly